src/compiler/checker.ts

```diff
@@ -32023,14 +32066,34 @@ namespace ts {
         }
 
         function checkInExpression(left: Expression, right: Expression, leftType: Type, rightType: Type): Type {
+            const privIdOnLeft = isPrivateIdentifier(left);
+            if (privIdOnLeft) {
+                if (languageVersion < ScriptTarget.ESNext) {
+                    checkExternalEmitHelpers(left, ExternalEmitHelpers.ClassPrivateFieldIn);
+                }
+                // Unlike in 'checkPrivateIdentifierExpression' we now have access to the RHS type
+                // which provides us with the oppotunity to emit more detailed errors
+                const symbol = getNodeLinks(left.parent).resolvedSymbol;
+                if (!symbol) {
+                    const isUncheckedJS = isUncheckedJSSuggestion(left, rightType.symbol, /*excludeClasses*/ true);
+                    reportNonexistentProperty(left, rightType, isUncheckedJS);
+                }
+            }
             if (leftType === silentNeverType || rightType === silentNeverType) {
                 return silentNeverType;
             }

```

Comment:
I’ve been meaning to look into the different between never and silentNever, thanks!

Reply:
I've moved this to the top now. I hadn't done that because I thought we wouldn't get the `checkExternalEmitHelpers` error when the RHS was `never` but I checked and as you suggested - this is still OK as the `silentNeverType` has a different use case.